Sources/AppHost/OsmAndAppImpl.mm

```diff
@@ -301,6 +302,25 @@ - (void) migrateResourcesToDocumentsIfNeeded
         _resourcesManager->rescanUnmanagedStoragePaths(true);
 }
 
+- (void)migrateOldWidgetKeysIfNeeded
```

Comment:
```
@objcMembers final class StorageMigrationManager: NSObject {
    enum MigrationKey: String {
        case changeWidgetIdsMigration1Key
    }
    
    static let shared = StorageMigrationManager()
    let defaults = UserDefaults.standard
    
    private override init() {}
    
    func changeWidgetIdsMigration1() {
        guard defaults.object(forKey: MigrationKey.changeWidgetIdsMigration1Key.rawValue) == nil else { return }
        let changeWidgetIds = [
            "heartRate" : "ant_heart_rate",
            "bicycleCadence" : "ant_bicycle_cadence",
            "bicycleDistance" : "ant_bicycle_distance",
            "bicycleSpeed" : "ant_bicycle_speed",
            "temperature" : "temperature_sensor"
        ]
        // all logic of migration
        // ....
        defaults.set(true, forKey:  MigrationKey.changeWidgetIdsMigration1Key.rawValue)
    }
}
```

It's better to create a separate migration manager. The template is provided above